client/landing/domains/utils.js

```diff
@@ -2,17 +2,22 @@
  * External dependencies
  */
 import { get } from 'lodash';
-import { moment } from 'i18n-calypso';
+import { getLocaleSlug } from 'i18n-calypso';
+import moment from 'moment';
 
 export function getMaintenanceMessageFromError( error, translate ) {
 	const maintenanceEndTime = get( error, [ 'data', 'maintenance_end_time' ], null );
+	const localeSlug = getLocaleSlug();
 
 	if ( maintenanceEndTime ) {
 		return translate(
 			"Our domain management system is currently undergoing maintenance and we can't process your request right now. Please try again %(when)s.",
 			{
 				args: {
-					when: moment.unix( maintenanceEndTime ).fromNow(),
+					when: moment
+						.unix( maintenanceEndTime )
+						.locale( localeSlug )
+						.fromNow(),
```

Comment:
This won't be picking up locale changes when they happen -- that's what `withLocalizedMoment` does, and it's not used anywhere nearby.

The code that uses the `getMaintenanceMessageFromError` helper is in need of a major rewrite. The _input_ for this helper (the `error` object) should be in state, not the _output_. Locale can be different at the time of rendering and at the time when the error state is being set. And we want to use the locale that's active at the time of _rendering_.

Reply:
I completely agree that this is currently being done incorrectly, and this PR doesn't change it. Do you think it falls within the scope of this PR to fix it?

Reply:
Yes, I think we should use this opportunity to fix it. I can contribute with a few commits.